### DISCUSSION TITLE
Why is there no --out option in parcel 2?

### DISCUSSION THREAD

**Discussion by danieltroger** [parcel-bundler/parcel#5541] [Q&A] [answered]
Hello guys, I've finally updated to parcel 2.

Apparently the --out option was removed.

There is some way to put an outfile name into package.json instead, but that doesn't work in my case.

My problem is that the out file name when running `parcel serve` and `parcel build` is the same and therefore if I run build it will overwrite the dist/main.js file from `parcel serve` and `parcel serve` will serve me a production build.

I just debugged for 15 minutes why my environment variables was production in development and I think it was because of that. Is there any good reason it was removed? Any workarounds?

**Comment by mischnic** [comment, marked as answer]
A workaround:
```json
{
	"dev": "build/index.html",
	"prod": "dist/index.html",
	"scripts": {
		"start": "parcel serve index.html --target dev",
		"build": "parcel build index.html --target prod"
	}
}
```
An alternative: set `--dist-dir` to a different folder for each invocation so that they don't overlap.

<br>

The general idea behind removing it is that having multiple targets (e.g. main & module) and `--out` doesn't make sense, because they have to have different paths.

**Comment by danieltroger** [comment]
Hey, thanks a lot for the reply! What am I doing wrong?
```
$ npm run serve

> my-project@1 serve
> dotenv -e .env.development -- parcel serve --port 1234 --target dev src/main.ts

ℹ️ Server running at http://localhost:1234
🚨 Build failed.
@parcel/core: Could not find target with name "dev"
```

I added this above the scripts in config.json

```
  "dev": "dist/dev.js",
  "prod": "dist/prod.js",
  ```

**Comment by danieltroger** [comment]
I figured it out! I need to duplicate the section where I say what I want to build for and call it dev or prod

**Comment by danieltroger** [comment]
@mischnic I now have this:
```
  "dev": "my_filename.js",
    "scripts": {
      "serve": "dotenv -e .env.development -- parcel serve --port 1234 --target dev --dist-dir dev_dist src/main.ts"
      },
      "targets": {
        "dev": {
      "engines": {
        "browsers": "last 1 firefox version"
      }
    }
  }
 ```
 
When I run npm run serve it creates dev_dist and builds the file into it, like expected. But then it infinite loops: it uses high CPU and the "built in x ms" counter updates around 3x per second. Is this a bug?

**Reply by mischnic** [reply]
Probably the same bug as https://github.com/parcel-bundler/parcel/issues/5355
